gpflow/likelihoods/likelihoods.py

```diff
@@ -69,92 +69,304 @@ def inv_probit(x):
 
 
 class Likelihood(Module):
-    def __init__(self):
+    def __init__(self, latent_dim: int, observation_dim: int):
+        """
+        A base class for likelihoods, which specifies an observation model 
+        connecting the latent functions ('F') to the data ('Y').
+
+        All of the members of this class are expected to obey some shape conventions, as specified
+        by latent_dim and observation_dim.
+
+        If we're operating on an array of function values 'F', then the last dimension represents
+        multiple functions (preceding dimensions could represent different data points, or
+        different random samples, for example). Similarly, the last dimension of Y represents a
+        single data point. We check that the dimensions are as this object expects.
+
+        The return shapes of all functions in this class is the broadcasted shape of the arguments,
+        up to the last dimensions.
+
+        :param latent_dim: the dimension of the vector F of latent functions for a single data point
+        :param observation_dim: the dimension of the observation vector Y for a single data point
+        """
         super().__init__()
         self.num_gauss_hermite_points = 20
+        self.latent_dim = latent_dim
+        self.observation_dim = observation_dim
 
-    def predict_mean_and_var(self, Fmu, Fvar):
-        r"""
-        Given a Normal distribution for the latent function,
-        return the mean of Y
+    def check_last_dims_valid(self, F, Y):
```

Comment:
Do we expect users (someone using a likelihood without looking at its implementation - not someone who is writing a new likelihood) to ever call this directly ? What would be the use-case for that? if not, it would be good to mark them private
```suggestion
    def _check_last_dims_valid(self, F, Y):
```